src/dftbp/common/envcheck.F90

```diff
@@ -0,0 +1,75 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2022  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+#:include 'error.fypp'
+
+!> Contains functionality to check environment settings.
+module dftbp_common_envcheck
+
+  use, intrinsic :: iso_c_binding, only : c_char, c_int
+  use dftbp_common_accuracy, only : dp
+  use dftbp_common_environment, only : TEnvironment
+  use dftbp_common_globalenv, only : stdOut
+  use dftbp_io_message, only : warning
+  implicit none
+
+  private
+  public :: checkStacksize
+
+
+contains
+
+  !> Checks stacksize settings for optimal user experience/performance.
+  subroutine checkStacksize(env)
+
+    !> Environment settings
+    type(TEnvironment), intent(in) :: env
+
+    !! Current stacksize
+    integer :: cStack
+
+    !! Error status
+    integer :: iErr
+
+    interface
+      ! Interface for checking stacksize through C-routine.
+      subroutine get_stacksize(cStack, iErr) bind(C, name="get_stacksize_c")
+
+        import :: c_char, c_int
+        implicit none
+
+        !> Current stacksize
+        integer(kind=c_int), intent(out) :: cStack
+
+        !> Error status
+        integer(kind=c_int), intent(out) :: iErr
+
+      end subroutine get_stacksize
+    end interface
```

Comment:
Please move that in the preamble of the module.

Reply:
done